packages/opentelemetry-resources/src/platform/node/detectors/ProcessDetector.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  Detector,
+  Resource,
+  PROCESS_RESOURCE,
+  ResourceDetectionConfigWithLogger,
+} from '../../../';
+import { ResourceAttributes } from '../../../types';
+
+/**
+ * ProcessDetector will be used to detect the resoureces related current process running
+ * and being instumented from the NodeJS Process module.
+ */
+class ProcessDetector implements Detector {
+  async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
+    const processResource: ResourceAttributes = {
+      [PROCESS_RESOURCE.PID]: process.pid || '',
+      [PROCESS_RESOURCE.NAME]: process.title || '',
+      [PROCESS_RESOURCE.COMMAND]: process.argv[1] || '',
+      [PROCESS_RESOURCE.COMMAND_LINE]: process.argv.join(' ') || '',
```

Comment:
Can't this leak secrets with they are injected though command line ? does the spec have specific comment about that ?

Reply:
No the specification doesn't have any note on it, may be relying for these secrets over ENV variable ? 

Here is the process spec : https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/resource/semantic_conventions/process.md

Reply:
CLI can already leak secrets in so many ways. bash history, ps command, etc. I don't think it is typically recommended or expected that the command line contains secrets

Reply:
can this be optional then ?